packages/connect/src/types/settings.ts

```diff
@@ -35,4 +36,6 @@ export interface ConnectSettings {
     env: 'node' | 'web' | 'webextension' | 'electron' | 'react-native';
     timestamp: number;
     proxy?: Proxy;
+    // TODO: this approach can be used only when not serializing settings.
+    logWriter?: (message: LogMessage) => void;
```

Comment:
@mroz22 Being able to pass a `logWriter`  at connect init would be great but only for some environments.
Maybe in this PR we could focus on the share worker and then address other use cases.

Reply:
sure not needed to implement plugability right away. just something to think about for future